### PR TITLE
adds exdates, exdate, exrules and exrule as options for rrule with rr…

### PR DESCRIPTION
**Pull request by BoLaMN**
adds exdates, exdate, exrules and exrule as options for rrule with rruleset

implements #4439, #4549

### CODE REVIEW

**Comment by BoLaMN** [comment]
currently failing on 

```
● handles multiple prop changes, include event reset
    expect(received).toBe(expected) // Object.is equality
    Expected: 2
    Received: 1
      79 |     expect(isWeekendsRendered(wrapper)).toBe(false)
      80 |     expect(eventRenderCnt).toBe(3) // +2
    > 81 |     expect(viewSkeletonRenderCnt).toBe(2) // +1
         |                                   ^
      82 |   })
      83 | })
      84 | 
      at toBe (FullCalendar.test.js:81:35)
``` 

but not why my changes would be effecting it at all

**Comment by stunaz** [comment]
Any update on this? much appreciated

**Comment by BoLaMN** [comment]
yeah not sure either, been over a month now without any review. 

Im guessing @arshaw is a one man band with pull requests and has priorities else where.

**Comment by 21Junipers** [comment]
@arshaw makes it clear that PRs might take a while to implement for various (completely) reasons. I hope it does not offend to state here how much rruleset support would bring to the table. Wish I could be useful in some way to get this in sooner rather than later. NOT complaining, this lib has come so far with v4 (Thanks!).

File: packages/rrule/src/main.ts [hunk 1]
```diff
@@ -104,6 +123,20 @@ function parseRRule(input, dateEnv: DateEnv) {
   return null
 }
 
+function getExArray(input, key) {
+  let arr
+
+  if (input[key + 's']) {
```
Comment (arshaw) [inline packages/rrule/src/main.ts]: could you help me understand what's going on here? some comments would great.

Comment (BoLaMN) [reply packages/rrule/src/main.ts]: allows it to be defined as

exdate: new Date()
exdates: [ new Date(), new Date() ]
exrule: new Date()
exrules: []

in the options and transform it into an array on the return

**Comment by 21Junipers** [comment]
Dead?

**Comment by BoLaMN** [comment]
It’s waiting for tests to be written, unfortunately my free time is negative 7 hours a day so till I can get around to it or someone else wants to contribute tests it’ll probably sit here for awhile

**Comment by yalanyali** [comment]
This looks pretty good, I'd love to see it being published as well.

**Comment by gongAll** [comment]
Recently I upgraded from v4 to v5 and some features broke. Basically passing rrules as string would allow for exdates to be passed on v4 and won't be allowed on v5 anymore.

Therefore, I see this pr as really important, since I cannot pass my project with v5 to production without it, which is a bummer. Would be glad to make tests for it, having some guidelines to follow.

**Comment by arshaw** [comment]
Superseded by https://github.com/fullcalendar/fullcalendar/pull/5097, which has been released in v5.5.0. Please see my comment: https://github.com/fullcalendar/fullcalendar/issues/4439#issuecomment-748536028

**Pull request closed without merging**